dot/parachain/types/overseer_message.go

```diff
@@ -74,16 +86,79 @@ type ProspectiveParachainsMessageCandidateSeconded struct {
 	CandidateHash CandidateHash
 }
 
-// IntroduceCandidateRequest is a request to introduce a candidate into the Prospective Parachains Subsystem.
-type IntroduceCandidateRequest struct {
-	// The para-id of the candidate.
+// Get the hypothetical frontier membership of candidates with the given properties
+// under the specified active leaves' fragment trees.
+//
+// For any candidate which is already known, this returns the depths the candidate
+// occupies.
+type ProspectiveParachainsMessageGetHypotheticalFrontier struct {
+	HypotheticalFrontierRequest HypotheticalFrontierRequest
+	Ch                          chan HypotheticalFrontierResponse
+}
+
+// Request specifying which candidates are either already included
+// or might be included in the hypothetical frontier of fragment trees
+// under a given active leaf
+type HypotheticalFrontierRequest struct {
+	// Candidates, in arbitrary order, which should be checked for possible membership in fragment trees
+	Candidates []HypotheticalCandidate
+	// Either a specific fragment tree to check, otherwise all.
+	FragmentTreeRelayParent *common.Hash
+	// Only return membership if all candidates in the path from the root are backed.
+	BackedInPathOnly bool
+}
+
+type HypotheticalFrontierResponse []struct {
+	HypotheticalCandidate HypotheticalCandidate
+	Memberships           []FragmentTreeMembership
+}
```

Comment:
```suggestion
```suggestion
type HypotheticalFrontierResponse {
	HypotheticalCandidate HypotheticalCandidate
	Memberships           []FragmentTreeMembership
}

type HypotheticalFrontierResponses []HypotheticalFrontierResponse 
```
```